web-basics-browser-security-model/README.md

```diff
@@ -0,0 +1,403 @@
+
+Add the below HTML code in the same index.html as above:
+
+```
+<iframe src="sandbox_iframe.html" sandbox width=450 height=100>
+if you can see this, you browser doesn't understand SANDBOX IFRAME.
+</iframe>
+```
+
+Access the page via browser http://localhost.
+
+The value of the sandbox attribute can either be just sandbox (then all restrictions are applied), or a space-separated list of pre-defined values that will REMOVE the particular restrictions.
+
+# Same-Origin Policy
+
+The same-origin policy is a web browser security mechanism that aims to prevent websites from attacking each other. The same-origin policy restricts scripts on one origin from accessing data from another origin.
+
+## Why we need Same-Origin Policy?
+
+When a browser sends an HTTP request from one origin to another, any cookies, including authentication session cookies, relevant to the other domain are also sent as part of the request. This means that the response will be generated within the user's session, and include any relevant data that is specific to the user. Without the same-origin policy, if you visited a malicious website, it would be able to read your emails from Gmail and private messages from Facebook. You can read more about SOP here. [[5]](https://portswigger.net/web-security/cors/same-origin-policy)
+
+# Talking to web sites
+
+Let's go through the basics of how HTTP requests are made, using telnet to form requests. First, let's connect to the vulnerable web server:
+
+```
+user@hostname~#: telnet 141.85.224.157 80
+```
+
+Now let's issue a simple GET. The request is composed of:
+
+- GET <path> <http-version>
+- followed by other header contents
+- followed by an additional newline, indicating the end of the request.
+
+_Please note the above bullet points and the fact that you need to provide an additional newline to indicate the end of the request._
+
+```
+GET / HTTP/1.0
+```
+
+```
+HEAD / HTTP/1.0
+```
+
+```
+POST / HTTP/1.0
+```
+
+The server's response contains:
+
+- A status code (200 OK in our case)
+- Date information and information about the server
+- Encoding and other info about the data, i.e. it's MIME-type
+- The length of the data
+- The actual data
+
+# Tools
+
+## Postman
+
+Postman is a tool through which you can send HTTP requests. It can automate tests and integrate them into CI/CD pipeline. You can simulate endpoints to test without having to set up a backend. It can also monitor certain endpoints to check their health and performance. [[6]](https://www.postman.com/)
+
+## HTTPie
+
+HTTPie is also used to send HTTP requests. It can be used in the terminal, instead of utilities such as **curl** and **wget**. It has an easy syntax, comes with highlighting and JSON assets. [[7]](https://httpie.io/)
+
+## Hoppscotch
+
+Hoppscotch is a tool that you can use from your browser or as a browser extension to quickly test certain endpoints. It also assetss MQTT or GraphQL requests. [[8]](https://hoppscotch.io/)
```

Comment:
**assetss**?

Reply:
I don't know what was here but I changed with:
"""
It also supports MQTT or GraphQL requests.
"""